src/recipe/test/unit/connection.handler.gateway.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppModule } from '../../../app.module';
+import { AuthService } from '../../../auth/auth.service';
+import { LoggedUserPayload } from '../../../common/interfaces/logged-user-payload.interface';
+import { ConnectionHandlerGateway } from '../../../websocket/connection.handler.gateway';
+import { Socket } from 'socket.io-client';
+import { GatewayModule } from '../../../websocket/gateway.module';
+
+const verifiedToken: LoggedUserPayload = {
+  name: 'testName',
+  sub: 1,
+};
+
+const socket: Socket = {
+  id: 'testSocketId',
+  handshake: {
+    headers: {
+      authorization: 'testToken',
+    },
+  },
+};
+
+const invalidTokenError = new Error('invalid token');
+
+describe('Connection Handler Gateway', () => {
+  let connectionHandlerGateway: ConnectionHandlerGateway;
+  let authService: AuthService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [AppModule, GatewayModule],
+    }).compile();
+    await module.createNestApplication().init();
+
+    connectionHandlerGateway = module.get<ConnectionHandlerGateway>(
+      ConnectionHandlerGateway,
+    );
+    authService = module.get<AuthService>(AuthService);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    connectionHandlerGateway.connectedUsers.clear();
+  });
+
+  it('should add connected user to connectedUsers', async () => {
+    const authServiceVerifyToken = jest
+      .spyOn(authService, 'verifyToken')
+      .mockResolvedValue(verifiedToken);
+
+    const connectedUsersSet = jest.spyOn(
+      connectionHandlerGateway.connectedUsers,
+      'set',
+    );
+
+    await connectionHandlerGateway.handleConnection(socket);
+
+    const connectedUsers = connectionHandlerGateway.connectedUsers;
+
+    expect(authServiceVerifyToken).toBeCalledTimes(1);
+    expect(authServiceVerifyToken).toBeCalledWith(
+      socket.handshake.headers.authorization,
+    );
+    expect(connectedUsersSet).toBeCalledTimes(1);
+    expect(connectedUsersSet).toBeCalledWith(verifiedToken.sub, socket.id);
+    expect(connectedUsers.size).toBe(1);
+    expect(connectedUsers.get(verifiedToken.sub)).toBe(socket.id);
+  });
+
+  it('should throw an error when token is invalid while trying to add connected user to connectedUsers', async () => {
```

Comment:
This part of the test description:`trying to add a connected user to connectedUsers` is either confusing or incorrect. 
 By reading this I don't understand why an error should be thrown.

The description indicates that an error should be thrown but the test itself does not check that. 
you mock rejected value of `verifyToken` with an error.
What is your test in fact about is checking if a client is added to connectedUsers when the authService, 'verifyToken' rejects

Reply:
What i would do i would manipulate the handshare content and then test accordingly to check if proper errors were thrown or not